### PR TITLE
Add a step to scan workflow to be in DLQ

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Add a step to scan workflow to be in DLQ

**Why?**
Most of the case the DLQ message indicates a corrupted workflow in source cluster

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/replication/task_processor_test.go [hunk 1]
```diff
@@ -293,3 +300,46 @@ func (s *taskProcessorSuite) TestGenerateDLQRequest_ReplicationTaskTypeSyncActiv
...
+	}
+	fixExecution := entity.Execution{
+		DomainID:   domainID,
+		WorkflowID: workflowID,
+		RunID:      runID,
+		State:      persistence.WorkflowStateCorrupted,
+	}
+	jsArray, err := json.Marshal(fixExecution)
+	s.NoError(err)
+	requestMatcher := ctesting.NewMatcher(func(x interface{}) bool {
```
Comment (yycptt) [inline service/history/replication/task_processor_test.go:326]: nit: I sometimes use `DoAndReturn` to validate the inputs. 
https://github.com/uber/cadence/blob/58e8e1ccb65b95f2a99ed4940a52336e15ab2aa4/service/history/task/cross_cluster_task_processor_test.go#L149

File: service/history/replication/task_processor.go [hunk 2]
```diff
@@ -24,14 +24,20 @@ package replication
...
 	"strconv"
 	"sync/atomic"
 	"time"
 
+	"github.com/pborman/uuid"
 	"go.uber.org/yarpc/yarpcerrors"
 
+	"github.com/uber/cadence/common/reconciliation"
+	"github.com/uber/cadence/common/reconciliation/entity"
+
```
Comment (yycptt) [inline service/history/replication/task_processor.go:40]: nit: remove empty line

File: common/reconciliation/constants.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+package reconciliation
+
+// Execution fixer workflow relates
+
+const (
+	ExecutionFixerWorkflowType        = "execution-fixer-workflow"
+	ExecutionFixerWorkflowTaskList    = "execution-fixer-tl"
+	ExecutionFixerWorkflowSignalName  = "execution-fixer-signal"
+	ExecutionFixerWorkflowID          = "execution-fixer-workflow-id"
+	ExecutionFixerWorkflowTimeout     = 24 * 60 * 60
```
Comment (yycptt) [inline common/reconciliation/constants.go:30]: nit: probably indicate the unit of the timeout?

File: service/history/replication/task_processor.go [hunk 4]
```diff
@@ -540,6 +551,61 @@ func (p *taskProcessorImpl) generateDLQRequest(
...
+	default:
+		return nil
+	}
+	clusterName := p.shard.GetClusterMetadata().ClusterNameForFailoverVersion(failoverVersion)
+	client := p.shard.GetService().GetClientBean().GetRemoteFrontendClient(clusterName)
+	fixExecution := entity.Execution{
+		DomainID:   domainID,
+		WorkflowID: workflowID,
+		RunID:      runID,
+		State:      persistence.WorkflowStateCorrupted,
```
Comment (yycptt) [inline service/history/replication/task_processor.go:587]: nit: mind add a comment here that says we assume it's corrupted here and rely on the check before fix logic in the fixer workflow to decide if it's actually a corruption?

**Review by yycptt — APPROVED** [review]

File: service/history/replication/task_processor_test.go (hunk 1 above)
Comment (yux0) [reply service/history/replication/task_processor_test.go:326]: Good to know.

**Pull request merged**